### PR TITLE
IDEMPIERE-5723 NPE when allocate has both AR and AP invoice

### PR DESCRIPTION
**Pull request by CarlosRuiz-globalqss**
https://idempiere.atlassian.net/browse/IDEMPIERE-5723

Integrating patches from @hieplq for the JUnit test and Gaurav for the fix

# Pull Request Checklist

- [ ] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [ ] My code follows the best practices of this project
- [ ] I have performed a self-review of my own code
- [ ] My code is easy to understand and review. 
- [ ] I have checked my code and corrected any misspellings
- [ ] In hard-to-understand areas, I have commented my code.
- [ ] My changes generate no new warnings
### Tests
- [ ] I have tested the direct scenario that my code is solving
- [ ] I checked all collaterals that can be affected by my changes, and tested other potential affected scenarios
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by hieplq** [comment]
base on idea from @gauravsontakke i rewrite patch and make pull request [here](https://github.com/idempiere/idempiere/pull/1842)
i will comment on things i don't like on this pull request that why i re-write it

File: org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java [hunk 1]
```diff
@@ -190,6 +190,7 @@ public ArrayList<Fact> createFacts (MAcctSchema as)
 		Fact factForRGL = new Fact(this, as, Fact.POST_Actual); // dummy fact (not posted) to calculate Realized Gain & Loss
 		boolean isInterOrg = isInterOrg(as);
 		MAccount bpAcct = null;		//	Liability/Receivables
+		ArrayList<Integer> bpAccts = new ArrayList<Integer>();
```
Comment (hieplq) [inline org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java:193]: it look like we has many bpAcct as many invoice but actual we has 2 bpAcct for ar and ap only

File: org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java [hunk 2]
```diff
@@ -1091,12 +1105,12 @@ private String createInvoiceRoundingCorrection (MAcctSchema as, Fact fact, MAcco
 				.append(" FROM Fact_Acct ")
 				.append("WHERE AD_Table_ID=? AND Record_ID=?")
 				.append(" AND C_AcctSchema_ID=?")
-				.append(" AND Account_ID=?")
+				.append(" AND Account_ID in ( "+commaseparatedlist+" ) ")
```
Comment (hieplq) [inline org.adempiere.base/src/org/compiere/acct/Doc_AllocationHdr.java:1108]: old logic, each invoice need one account to query fact, so i like to keep that logic

**Comment by CarlosRuiz-globalqss** [comment]
Closing as replaced by https://github.com/idempiere/idempiere/pull/1842

**Pull request closed without merging**